### PR TITLE
Fix #101 - throw an error if the x argument supplied to vis_dat is not a data.frame

### LINKED ISSUE #101: More informative error message when user supplies non data.frame object to vis_dat
I was trying out visdat::vis_dat for the first time (great package!) and supplied datasets::AirPassengers as the x argument (I mistakenly thought it was a data frame and not a time series).

` visdat::vis_dat(AirPassengers)`

The error returned was:

> Error in if (ncol(x) * nrow(x) > large_data_size && warn_large_data) { : 
  missing value where TRUE/FALSE needed

I was wondering if it would be worth having an error message which warns the user that they should be supplying a data.frame object?

(Happy to submit a PR myself if you'd like)

### PR DESCRIPTION
**Pull request by thisisnic**
I have updated vis_dat to include a check for if x is an object of class data.frame, and if not throws error "x must be a data.frame object".

### CODE REVIEW

File: R/vis-dat.R [hunk 1]
```diff
@@ -64,6 +64,10 @@ vis_dat <- function(x,
                     warn_large_data = TRUE,
                     large_data_size = 900000) {
 
+  if(!("data.frame" %in% class(x))){
```
Comment (njtierney) [inline R/vis-dat.R]: Perfect! Can I suggest that you use 

```r
! inherits(x, "data.frame)
```

**Comment by thisisnic** [comment]
Thanks for your help and feedback! :)

I've made the updates suggested above (using inherits(), a more user-friendly error message, and adding a test).

I wasn't sure if the error message I added is a little verbose in phrasing - what do you think?

In the example of calling vis_dat(AirPassengers), it would return:
 `Error: Oops - vis_dat requires x to be an object of class data.frame but the object you've provided me with is of class(es): ts`

**Comment by njtierney** [comment]
Fantastic, this looks excellent - well done! :)

So writing error messages is hard, and I think that this is great! A few minor changes, I would suggest something like:

`vis_dat requires a data.frame, but currently, the object I see has the class/es: class(x)`

I changed `class(es)` because I initially thought that the function `class` was being applied on the object `es`. 

What do you think? The most important thing is to think about your intial point - the initial error message from `visdat` wasn't very clear or helpful - do you think that if you saw this error message that you would know what to do to fix it? To me, if an error message can indicate what to do to fix the problem, then that's a win.

### Extending this across other parts of visdat.

So, this is actually a really useful feature, that I think should be in all the `vis_` functions of visdat!

What do you think about writing this up as a function that can be called in other functions in visdat?

This line of thinking sounded familiar, and it turns out that I have done this in naniar with a function called [`test_if_dataframe()`](https://github.com/njtierney/naniar/blob/master/R/utils.R#L100), which is then called in placed like [this](https://github.com/njtierney/naniar/blob/20952b55ad505b2a7cb3563bbc78bf2a347c556c/R/impute-median.R#L98).

So, this could be the first line in all the `vis_*` family of functions.
 
Of course, I am happy to do this, you have already made a very useful contribution here, but I just thought I'd see if you wanted to continue?

In any case, I'm so glad that you filed the issue #101, I think I'll be more aware of error messages in the future. :)

**Comment by thisisnic** [comment]
Thanks again for taking the time to go through this with me!

Yep - I like the phrasing there as it explains everything the end user needs to know without being too wordy or complex.  I've updated the code in my PR to include this.

Sounds great, I'd love to!  Shall I create a utils.R file in this repo and copy the function across from naniar?  Or perhaps it'd fit into internals.R?  Or something else?  Let me know! :)

File: R/vis-dat.R [hunk 2]
```diff
@@ -66,7 +66,7 @@ vis_dat <- function(x,
 
   # throw error if x not data.frame
   if(!(inherits(x, "data.frame"))){
-    stop("Oops - vis_dat requires x to be an object of class data.frame but the object you've provided me with is of class(es): ",
+    stop("vis_dat requires a data.frame but the object I see has class/es: ",
          paste(class(x), collapse = ", "))
```
Comment (njtierney) [inline R/vis-dat.R]: Can you add the option for `stop`, `call. = FALSE`, as is done in the [`naniar` function](https://github.com/njtierney/naniar/blob/master/R/utils.R#L59) :)

**Comment by njtierney** [comment]
No problems at all - thank _you_ for doing all the work!

PR looks great! I've made a minor comment about using `call. = FALSE` - this means that the call isn't reported with the error message, I thinks this is a pattern I borrowed from some tidyverse packages. 

Good question about where to put it - could you put it in the `internals.R` file?

Thanks again for your help!

Oh, and make sure to add a bullet to the news.md file to document your great work and contribution, and also add yourself as a contributor in the DESCRIPTION, :) Thanks!

**Comment by thisisnic** [comment]
Awesome; I've added the function to internals.R, implemented it in all the main vis_* function, added the relevant test for each function, and added myself to DESCRIPTION.  Let me know if there's any other tweaks that need making!

**Comment by njtierney** [comment]
Wonderful! Thank you very much for this, @thisisnic !
